include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Gianvito Urgese <gianvito.urgese AT polito.it>
+ * \author Emanuele Parisi <emanuele.parisi AT polito.it>
+ * \brief Contains the format_ctd struct and its helper functions.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <regex>
+
+#include <seqan3/argument_parser/detail/format_base.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief The format that prints the Common Tool Descriptor file to std::cout.
+ * \ingroup argument_parser
+ *
+ * \details
+ *
+ * The CTD file is not written immediately, because the whole DOM tree
+ * composing the XML document can be completely built only after the parser
+ * is completely initialized. Instead, every call is stored and evaluated only
+ * when format_ctd::parse() is called.
+ */
+class format_ctd : format_base
+{
+
+public:
+    
+    /*!\brief Adds calls for appending 'clielement' and 'ITEM' nodes to the CTD DOM tree to be evaluated at parse time.
+     *
+     * \tparam option_type    The type of variable in which to store the given command line argument.
+     * \tparam validator_type The type of validator applied to the value after parsing.
+     *
+     * \param[out] value     The variable in which to store the given command line argument.
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
+     * \param[in]  desc      The description of the option.
+     * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
+     * \param[in]  validator The validator applied to the value after parsing (callable).
+     */
+    template<typename option_type, typename validator_type>
+    void add_option(option_type & value,
+                    char const short_id,
+                    std::string const & long_id,
+                    std::string const & desc,
+                    option_spec const & spec,
+                    validator_type && validator) 
+    {
+        // TODO (emanueleparisi) The current version of the CTD exporter does not support list options.
```

Comment:
Did you plan to do it in this PR or another? Both would be ok

Reply:
That's an issue ! If I'm not missing something, it would be a problem to implement CTD list options with the current argument parser... The reason is that the number of values for an 'option list' are defined only at runtime, while the CTD would required the number of values for a list option to be fixed. Am I missing something ? Should I open a separate issue for better discussing this point ?